src/hidimstat/conditional_permutation_importance.py

```diff
@@ -67,14 +68,11 @@ def __init__(
             n_jobs=n_jobs,
             n_permutations=n_permutations,
         )
-        self.rng = np.random.RandomState(random_state)
         self._list_imputation_models = []
-
-        self.imputation_model = {
-            "continuous": imputation_model_continuous,
-            "categorical": imputation_model_categorical,
-        }
         self.categorical_max_cardinality = categorical_max_cardinality
+        self.imputation_model_categorical = imputation_model_categorical
+        self.imputation_model_continuous = imputation_model_continuous
+        self.random_state = check_random_state(random_state)
```

Comment:
Can you do the check in the fit method and not in the __init__ method?
This is mostly done like this in scikitlearn.

Reply:
yes, done